### PR TITLE
Update javadoc

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

File: src/main/java/com/n1analytics/paillier/EncryptedNumber.java [hunk 1]
```diff
@@ -18,73 +18,7 @@
 import java.math.BigInteger;
 
 /**
- * Immutable class representing encrypted number and arithmetic operations that can be computed on the encrypted number.
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/EncryptedNumber.java]: This change seems to remove lots of clear high level documentation about an EncryptedNumber.

Comment (mpnd) [inline src/main/java/com/n1analytics/paillier/EncryptedNumber.java]: I've updated the high level documentation in 241cba3ff1395a573509067d2d56bb5185817a67

File: src/main/java/com/n1analytics/paillier/PaillierContext.java [hunk 2]
```diff
@@ -362,22 +680,59 @@ public EncryptedNumber encrypt(EncodedNumber encoded) {
...
+   */
   public EncryptedNumber encrypt(long value) {
     return encrypt(encode(value));
   }
 
+  /**
+   * Adds two EncryptedNumbers. Checks whether the {@code PaillierContext} of {@code operand1}
+   * and {@code operand2} are the same as this {@code PaillierContext}. If the operands' exponents
+   * are not the same, reduce the higher exponent to match with the lower exponent. Throws
+   * PaillierContextMismatchException if the {@code PaillierContext} of either
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/PaillierContext.java]: Don't need to repeat this statement in the summary. The `@throws` clause should say when the exception will be thrown.

File: src/main/java/com/n1analytics/paillier/PaillierContext.java [hunk 3]
```diff
@@ -397,18 +752,53 @@ public EncryptedNumber add(EncryptedNumber operand1, EncryptedNumber operand2)
...
+   */
   public EncryptedNumber add(EncryptedNumber operand1, EncodedNumber operand2)
           throws PaillierContextMismatchException {
     return add(operand1, encrypt(operand2));
   }
 
+  /**
+   * Adds an {@code EncodedNumber} and an {@code EncryptedNumber}. Encrypts the {@code EncodedNumber}
+   * before adding them together. Throws PaillierContextMismatchException if the {@code PaillierContext}
+   * of either {@code operand1} or {@code operand2} does not match this {@code PaillierContext}.
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/PaillierContext.java]: I think this is slightly wrong. The `PaillierContextMismatchException` won't get thrown unless the contexts can't be *made* to match - they don't have to be the same initially though?

File: src/main/java/com/n1analytics/paillier/PaillierContext.java [hunk 4]
```diff
@@ -432,6 +822,16 @@ public EncodedNumber add(EncodedNumber operand1, EncodedNumber operand2)
...
   }
 
+  /**
+   * Returns the additive inverse of {@code EncryptedNumber}. Throws PaillierContextMismatchException
+   * if the {@code PaillierContext} of either {@code operand1} or {@code operand2} does not match this
+   * {@code PaillierContext}.
+   *
+   * @param operand1 input.
+   * @return the additive inverse result.
+   * @throws PaillierContextMismatchException if the {@code PaillierContext} of {@code operand1}
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/PaillierContext.java]: Same as last comment here.

File: src/main/java/com/n1analytics/paillier/PaillierPrivateKey.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 import java.security.SecureRandom;
 
 /**
- * Immutable class representing Paillier private key.
+ * A class representing Paillier private key.
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/PaillierPrivateKey.java]: Is a PrivateKey now mutable?

File: src/main/java/com/n1analytics/paillier/PaillierPrivateKey.java [hunk 6]
```diff
@@ -90,21 +90,59 @@
...
+  /**
+   * The corresponding public key.
+   */
   protected final PaillierPublicKey publicKey;
-  protected final BigInteger p; //p and q are the two primes such that 
+
+  /**
+   * The first prime number, {@code p} such that {@code p*q = publicKey.modulus}.
+   */
+  protected final BigInteger p; //p and q are the two primes such that
```
Comment (hardbyte) [inline src/main/java/com/n1analytics/paillier/PaillierPrivateKey.java]: Remove the inline comment as it is now in the javadoc

**Comment by hardbyte** [comment]
Other than a few minor things like repeating the same sentences this gets a :+1: from me.

An overview document would be great as well, but if you make the small changes I'll get this merged in first.

**Pull request merged**